osp/core/ontology/installation.py

```diff
@@ -23,12 +25,40 @@ def __init__(self, namespace_registry=None, path=None):
         """
         self.namespace_registry = namespace_registry
         self.path = path
-        if self.path is None:
-            import osp.core.namespaces as namespaces
-            self.path = namespaces._path
         if self.namespace_registry is None:
-            import osp.core.namespaces as namespaces
-            self.namespace_registry = namespaces._namespace_registry
+            from osp.core.ontology.namespace_registry import namespace_registry
+            self.namespace_registry = namespace_registry
+
+    @property
+    def path(self) -> str:
+        """Path where ontologies are installed."""
+        return self._path or self.get_default_installation_path()
```

Comment:
The problem with doing this in the constructor is that then the default installation path cannot be changed (it would be fixed at `__init__`time).

Using a property, the default installation path is changed for every `OntologyInstallationManager` that does not have a custom path.

Reply:
You can still use a property, but rather than doing that check every time, you can do it once in the constructor?

Reply:
I just meant
```py
class OntologyInstallationManager:

    def __init__(self, namespace_registry=None, path=None):
        self.namespace_registry = namespace_registry
        self.path = path or self.get_default_installation_path()
        if self.namespace_registry is None:
            from osp.core.ontology.namespace_registry import namespace_registry
            self.namespace_registry = namespace_registry

    @property
    def path(self) -> str:
        return self._path

    @path.setter
    def path(self, value: str):
        self._path = value
```

It's a minor difference, but you avoid one check if the path property is queried often. I guess we could run into issues if we change the default path at some point, but that would anyway be a breaking change.

Reply:
> I guess we could run into issues if we change the default path at some point.

This is the thing I precisely want to avoid. Honestly I don't see `pico` as a critical element in terms of performance (I don't see people having to call `pico` thousands of times per second). For me, the balance between performance and consistency in this case is on the consistency side.

Besides, how am I supposed to test this functionality if I cannot change the default path? Remember that I only introduced this for testing and that the path is fixed as soon as I import `install`, `uninstall`, `packages` or `namespaces` from `osp.core.pico`. That is more related to [this other comment](https://github.com/simphony/osp-core/pull/761/files/9fbfe4dee86dfbf4494593e0cdc5cfd5194dab2e#r825761345) anyway.

Moreover, allowing to change the default path is **not** a breaking change, but a backwards compatible change. Existing code will always work, because it never changes it. I would say that it is even a _"bugfix"_. Remember that you can set the default path already using the `OSP_ONTOLOGIES` [environment variable](https://simphony.readthedocs.io/en/dev/utils.html?highlight=OSP_ONTOLOGIES#ontology-installation-folder). I think it is a plus if somebody sets `os.environ['OSP_ONTOLOGIES_DIR'] = 'new_path'`, and it works as expected (even though this PR is not enough to achieve so though, but goes in this direction).